libs/datamodel/core/tests/types/cockroachdb_native_types.rs

```diff
@@ -1,110 +1,237 @@
-        let native_type = &format!("{}(-1)", tpe);
-        test_native_types_without_attributes(native_type, "DateTime", &error_msg(native_type), COCKROACHDB_SOURCE);
-    }
+    let schema = indoc! {r#"
+        datasource db {
+          provider = "cockroachdb"
+          url      = env("DATABASE_URL")
+        }
+
+        model User {
+          id  Int      @id
+          val DateTime @db.Time(7)
+        }
+    "#};
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mArgument M is out of range for native type `Time(7)` of CockroachDB: M can range from 0 to 6.[0m
+          [1;94m-->[0m  [4mschema.prisma:8[0m
+        [1;94m   | [0m
+        [1;94m 7 | [0m  id  Int      @id
+        [1;94m 8 | [0m  val DateTime @[1;91mdb.Time(7)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expect_error(schema, &expectation);
+
+    let schema = indoc! {r#"
+        datasource db {
+          provider = "cockroachdb"
+          url      = env("DATABASE_URL")
+        }
+
+        model User {
+          id  Int      @id
+          val DateTime @db.Time(-1)
+        }
+    "#};
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mArgument M is out of range for native type `Time(-1)` of CockroachDB: M can range from 0 to 6.[0m
+          [1;94m-->[0m  [4mschema.prisma:8[0m
+        [1;94m   | [0m
+        [1;94m 7 | [0m  id  Int      @id
+        [1;94m 8 | [0m  val DateTime @[1;91mdb.Time(-1)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expect_error(schema, &expectation);
+
+    let schema = indoc! {r#"
+        datasource db {
+          provider = "cockroachdb"
+          url      = env("DATABASE_URL")
+        }
+
+        model User {
+          id  Int      @id
+          val DateTime @db.Timestamp(7)
+        }
+    "#};
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mArgument M is out of range for native type `Timestamp(7)` of CockroachDB: M can range from 0 to 6.[0m
+          [1;94m-->[0m  [4mschema.prisma:8[0m
+        [1;94m   | [0m
+        [1;94m 7 | [0m  id  Int      @id
+        [1;94m 8 | [0m  val DateTime @[1;91mdb.Timestamp(7)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expect_error(schema, &expectation);
+
+    let schema = indoc! {r#"
+        datasource db {
+          provider = "cockroachdb"
+          url      = env("DATABASE_URL")
+        }
+
+        model User {
+          id  Int      @id
+          val DateTime @db.Timestamp(-1)
+        }
+    "#};
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mArgument M is out of range for native type `Timestamp(-1)` of CockroachDB: M can range from 0 to 6.[0m
+          [1;94m-->[0m  [4mschema.prisma:8[0m
+        [1;94m   | [0m
+        [1;94m 7 | [0m  id  Int      @id
+        [1;94m 8 | [0m  val DateTime @[1;91mdb.Timestamp(-1)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expect_error(schema, &expectation);
```

Comment:
nitpick: in tests like this, it would be possible to test both errors in one schema to save space (please don't change this, just wanted to remark)